test/integration/test___main__.py

```diff
@@ -42,30 +41,26 @@ def test_main_bad_private_data_dir():
     cmdline('run', tmpfile, '-p', 'fake')
 
     try:
-        with raises(OSError):
+        with pytest.raises(OSError):
             main()
     finally:
         os.remove(tmpfile)
 
 
-def run_role(options, private_data_dir=None, expected_rc=0):
-    try:
-        private_data_dir = private_data_dir or tempfile.mkdtemp()
-        args = ['run', private_data_dir]
-        args.extend(options)
+def run_role(options, mocker, private_data_dir, expected_rc=0):
```

Comment:
You could turn `run_role` into a pytest fixture, to avoid passing `mocker` (a pytest fixture) as an argument to it.

Reply:
I tried that and the issue is we need to pass args to it. I _could_ make it a fixture by putting a function within the fixture, but decided to just pass in `mocker`. I don't care for local functions, which is what you have to do to make fixtures accept args.

Reply:
Wow. I also just realized the `args` here never gets passed to `main()`. 🤦‍♂️

Reply:
It is a very common pytest practice to define a function locally and return it. If you want to just define the method outside of the pytest fixture structure, then that's a perfectly fine preference to have. But in that case, using the mock from `from unittest import mock` makes more sense.

Directly defining the method in the test file but expecting a pytest `mocker` fixture to be passed is mixing the 2 worlds.

Reply:
>

Reply:
Getting this to accomplish the original intent of the test took quite a bit more work than changing this to a fixture. I ended up deleting this entirely.